lib/authenforce/middleware.go

```diff
@@ -1,21 +1,72 @@
 package authenforce
 
 import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"io/ioutil"
 	"net/http"
+	"sync/atomic"
 
+	"github.com/cuvva/cuvva-public-go/lib/cher"
 	"github.com/cuvva/cuvva-public-go/lib/crpc"
 	"github.com/wearemojo/mojo-public-go/lib/authparsing"
+	"github.com/wearemojo/mojo-public-go/lib/errgroup"
+	"github.com/wearemojo/mojo-public-go/lib/merr"
 )
 
-func (e Enforcer) CRPCMiddleware() crpc.MiddlewareFunc {
+func (e Enforcers) CRPCMiddleware() crpc.MiddlewareFunc {
 	return func(next crpc.HandlerFunc) crpc.HandlerFunc {
 		return func(res http.ResponseWriter, req *crpc.Request) error {
-			state := authparsing.GetAuthState(req.Context())
-			err := e(state)
+			ctx := req.Context()
+			authState := authparsing.GetAuthState(ctx)
+			mapReq := map[string]any{}
```

Comment:
Not sure what's best here - if the request has no body, would it be better for the map to be nil?